filters/openpolicyagent/openpolicyagent.go

```diff
@@ -476,38 +477,74 @@ func (registry *OpenPolicyAgentRegistry) new(store storage.Store, configBytes []
 // Start asynchronously starts the policy engine's plugins that download
 // policies, report status, etc.
 func (opa *OpenPolicyAgentInstance) Start(ctx context.Context, timeout time.Duration) error {
-	err := opa.manager.Start(ctx)
+	discoveryPlugin := discovery.Lookup(opa.manager)
 
-	if err != nil {
-		return err
-	}
+	done := make(chan struct{})
+	failed := make(chan error, 1)
 
-	// check readiness of all plugins
-	pluginsReady := func() bool {
-		for _, status := range opa.manager.PluginStatus() {
-			if status != nil && status.State != plugins.StateOK {
-				return false
+	discoveryPlugin.RegisterListener("startuplistener", func(status bundle.Status) {
+		handleStatusErrors(status, failed, "discovery plugin")
```

Comment:
I wonder if this should also be wrapped in a `sync.Once`. We cannot [currently unregister](https://github.com/open-policy-agent/opa/pull/6851) the listener and if it somehow triggers multiple times, it will write to a closed channel which will panic.